test/VaultRegistry.test.ts

```diff
@@ -1,445 +1,809 @@
+                    );
+                });
+            });
+
+            describe("when owner address is zero", () => {
+                it("reverts", async () => {
+                    await withSigner(
+                        this.erc20VaultGovernance.address,
+                        async (s) => {
+                            await expect(
+                                this.subject
+                                    .connect(s)
+                                    .registerVault(
+                                        this.anotherERC20Vault.address,
+                                        ethers.constants.AddressZero
+                                    )
+                            ).to.be.reverted;
+                        }
+                    );
+                });
+            });
+        });
+    });
+
+    describe("#stageProtocolGovernance", () => {
+        it("stages new ProtocolGovernance for commit", async () => {
+            await this.subject
+                .connect(this.admin)
+                .stageProtocolGovernance(this.newProtocolGovernance.address);
+
+            expect(await this.subject.stagedProtocolGovernance()).to.be.equal(
+                this.newProtocolGovernance.address
+            );
+        });
+        it("sets the stagedProtocolGovernanceTimestamp after which #commitStagedProtocolGovernance can be called", async () => {
+            await this.subject
+                .connect(this.admin)
+                .stageProtocolGovernance(this.newProtocolGovernance.address);
+
+            expect(
+                Number(await this.subject.stagedProtocolGovernanceTimestamp()) -
+                    now() -
+                    Number(await this.protocolGovernance.governanceDelay())
+            ).to.be.lessThanOrEqual(1);
+        });
+
+        describe("access control:", () => {
+            it("allowed: ProtocolGovernance Admin", async () => {
+                await this.subject
+                    .connect(this.admin)
+                    .stageProtocolGovernance(
+                        this.newProtocolGovernance.address
+                    );
+            });
+            it("denied: any other address", async () => {
+                let randomAddr = randomAddress();
+                while (randomAddr == this.admin.address) {
+                    randomAddr = randomAddress();
+                }
+                await withSigner(randomAddr, async (s) => {
+                    await expect(
+                        this.subject
+                            .connect(s)
+                            .stageProtocolGovernance(
+                                this.newProtocolGovernance.address
+                            )
+                    ).to.be.reverted;
+                });
+            });
+        });
+
+        describe("edge cases", () => {
+            describe("when new ProtocolGovernance is a zero address", () => {
+                it("does not fail", async () => {
```

Comment:
I think it should revert with `Exceptions.ADDRESS_ZERO` here. If there is no such check than it have to be added

Reply:
Imagine we've committed zero address as new ProtocolGovernance
then we'd face authorization issues
```solidity
    function _isProtocolAdmin(address sender) internal view returns (bool) {
        return _protocolGovernance.isAdmin(sender);
    }
```
Hence, we'd no longer be able to update ProtocolGovernance

Reply:
Agree